src/gt4py/next/program_processors/codegens/gtfn/itir_to_gtfn_ir.py

```diff
@@ -347,55 +399,20 @@ def visit_FunCall(self, node: itir.FunCall, **kwargs: Any) -> Node:
                     lhs=self.visit(node.args[0], **kwargs),
                     rhs=self.visit(node.args[1], **kwargs),
                 )
-            elif node.fun.id == "if_":
-                assert len(node.args) == 3
-                return TernaryExpr(
-                    cond=self.visit(node.args[0], **kwargs),
-                    true_expr=self.visit(node.args[1], **kwargs),
-                    false_expr=self.visit(node.args[2], **kwargs),
-                )
-            elif node.fun.id == "cast_":
-                assert len(node.args) == 2
-                return CastExpr(
-                    obj_expr=self.visit(node.args[0], **kwargs),
-                    new_dtype=self.visit(node.args[1], **kwargs),
-                )
-            elif node.fun.id == "list_get":
-                # should only reach this for the case of an external sparse field
-                return FunCall(
-                    fun=SymRef(id="tuple_get"),
-                    args=[self.visit(node.args[0]), self.visit(node.args[1])],
-                )
-            self._error_on_illegal_function_calls(node)
-            if node.fun.id == "cartesian_domain":
-                sizes, domain_offsets = self._make_domain(node)
-                return CartesianDomain(tagged_sizes=sizes, tagged_offsets=domain_offsets)  # type: ignore
-            elif node.fun.id == "unstructured_domain":
-                sizes, domain_offsets = self._make_domain(node)
-                connectivities = []
-                if "stencil" in kwargs:
-                    shift_offsets = self._collect_offset_or_axis_node(
-                        itir.OffsetLiteral, kwargs["stencil"]
-                    )
-                    for o in shift_offsets:
-                        if o in self.offset_provider and isinstance(
-                            self.offset_provider[o], common.Connectivity
-                        ):
-                            connectivities.append(SymRef(id=o))
-                return UnstructuredDomain(
-                    tagged_sizes=sizes,
-                    tagged_offsets=domain_offsets,
-                    connectivities=connectivities,
-                )  # type: ignore
+            elif hasattr(self, f"_visit_{node.fun.id}"):
+                # special handling of applied builtins is handled in `_visit_<builtin>`
+                return getattr(self, f"_visit_{node.fun.id}")(node, **kwargs)
```

Comment:
Code duplication for the f-string.
Pattern duplication for `hasattr`/`getattr`.
Could possibly be:
```python
visitor_method = getattr(self, f"_visit_{node.fun.id}", None)
if visitor_method:
  return visitor_method(node, **kwargs)
```
But it's not a big thing, I'm just nitpicking.

Reply:
resolved with walrus